### PR TITLE
Follow up install doc improvements

### PR DESCRIPTION
**Pull request by abrennan89**
Following up some comments from #3297 

- Removed some fluff (links should go in context docs not procedures, distracting)
- Updated kube min version
- `--filename` -> `-f` for consistency
- Removed install from source, belongs in dev docs

### CODE REVIEW

File: docs/install/knative-with-operators.md [hunk 1]
```diff
@@ -12,39 +12,24 @@ for development or test purposes only.
...
 - If you have multiple nodes in your cluster, for each node you will need at least 2 CPUs, 4 GB of memory, and 20 GB of disk storage.
-- Your Kubernetes cluster must have access to the internet, since the Knative Operator must download images.
+- Your Kubernetes cluster must have access to the internet, since Kubernetes needs to be able to fetch images, such as `gcr.io/knative-releases/knative.dev/operator/cmd/operator:<version>`.
+<!--TODO: Verify these requirements-->
 - You have installed [Istio](./installing-istio.md).
 
 ## Installing the latest release
 
-You can find the release information about the Knative Operator on the [Releases page](https://github.com/knative/operator/releases).
+You can information about the different released versions of the Knative Operator on the [Releases page](https://github.com/knative/operator/releases).
```
Comment (julz) [inline docs/install/knative-with-operators.md:25]: missing a verb after 'can' I think

File: docs/install/any-kubernetes-cluster.md [hunk 2]
```diff
@@ -24,25 +24,25 @@ You can install Knative by applying YAML files using the `kubectl` CLI.
...
 - If you have multiple nodes in your cluster, for each node you will need at least 2 CPUs, 4 GB of memory, and 20 GB of disk storage.
+<!--TODO: Verify these requirements-->
+- Your Kubernetes cluster must have access to the internet, since Kubernetes needs to be able to fetch images.
 
 ## Installing the Serving component
 
-1. Install the
-   [Custom Resource Definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
-   (aka CRDs):
+1. Install the required custom resources:
```
Comment (julz) [inline docs/install/any-kubernetes-cluster.md:36]: I think you still need the word 'definitions' here to distinguish from an instance of the CRD, but I may well be being too pedantic and noone cares 😅.

```suggestion
1. Install the required custom resource definitions:
```

**Review by julz — COMMENTED** [review]
lgtm with a couple tiny nits

File: docs/install/knative-with-operators.md (hunk 1 above)
Comment (abrennan89) [reply docs/install/knative-with-operators.md:25]: Oops! Thank you :slightly_smiling_face:

File: docs/install/any-kubernetes-cluster.md (hunk 2 above)
Comment (abrennan89) [reply docs/install/any-kubernetes-cluster.md:36]: I wondered this, however, in the kube docs, they seem to use CustomResourceDefinitions more specifically when talking about the API, and custom resources more as the thing that's actually installed?
https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/

I've been trying to nail down for a while when we should refer to what thing. If you think this is important to say though I'm OK with that, I think in general though we maybe need to find a way to make it clearer across docs the difference and when we mean CRs vs CRDs?

Comment (abrennan89) [reply docs/install/any-kubernetes-cluster.md:36]: I guess maybe it should be CustomResourceDefinition unless we mean an instance of a custom resource like it is in the kube docs, I just hate things being formatted this way in docs :sob: does it make sense to change it to CustomResourceDefinition though?

Comment (abrennan89) [reply docs/install/any-kubernetes-cluster.md:36]: Imma say "CustomResourceDefinition object" everywhere I think for this...

Comment (julz) [reply docs/install/any-kubernetes-cluster.md:36]: hah, no, I'm convinced by the argument that the Kube docs say the CR is the thing installed by the CRD and that this is, in fact, a total bike shed. Let's go with your simpler version :D.

Comment (abrennan89) [reply docs/install/any-kubernetes-cluster.md:36]: So...which one? You know more than me so if you make a call I'm happy to go with it :) sorry just not sure which is the "simpler version you mean" :joy:

Comment (julz) [reply docs/install/any-kubernetes-cluster.md:36]: I personally think, reflecting on this and the fact it seems like a hellmouth to the bike-shed dimension, we should just go with the one with less words. So I vote "custom resources". If anyone _really_ cares that this is technically wrong (Im not even sure it is now, honestly 😅) we can always change it. I regret bringing it up 😂.

**Review by julz — APPROVED** [review]
/lgtm

**Pull request merged**